lib/views/mobile/receive/receive.dart

```diff
@@ -83,11 +37,12 @@ class _ReceiveState extends State<Receive> {
           child: Container(
             key: Key(RECEIVE_SCREEN_BODY),
             padding: EdgeInsets.symmetric(horizontal: 8.0),
-            child: received == true
+            child: currentState == ReceiveScreenStates.FileReceived
                 ? ReceivingDone(fileSize, fileName)
                 : Container(
-                    child: isReceiving
-                        ? ReceiveProgress(fileSize, fileName)
+                    child: currentState == ReceiveScreenStates.FileReceiving
```

Comment:
A switch statement here can improve readability; You can call a function that takes `ReceiveScreenStates` and returns a `Widget` and maybe keep that function in `ReceiveShared`.

It's also possible to use a `Map` from `ReceiveScreenStates` to `Widget Function()` and lookup with some default value like an "under construction" Widget maybe, then call the value.